test/constraint/encoding.test.ts

```diff
@@ -26,6 +27,65 @@ describe('constraints/encoding', () => {
     }
   });
 
+  describe('hasAllRequiredPropertiesSpecific', () => {
+    let encModel = new EncodingConstraintModel(
+      {
+        name: 'Test EncodingModel for hasAllRequiredProperties class method',
+        description: 'Test EncodingModel for hasAllRequirdProperties class method',
+        properties: [Property.AGGREGATE, Property.TYPE, Property.SCALE, Property.SCALE_TYPE],
+        requireAllPropertiesSpecific: true,
+        strict: true,
+        satisfy: undefined
+      }
+    );
+
+    let encQ1: EncodingQuery = {
```

Comment:
You never re-use this encQ1 for multiple test cases.  Why don't you put it inside relevant `it()` so that you can name it `encQ` without `1`.   (Same for all others)